worldedit/transformations.lua

```diff
@@ -330,20 +396,17 @@ function worldedit.orient(pos1, pos2, angle)
 				local node = get_node(pos)
 				local def = registered_nodes[node.name]
 				if def then
-					local paramtype2 = def.paramtype2
-					if paramtype2 == "wallmounted" or
-							paramtype2 == "colorwallmounted" then
-						local orient = node.param2 % 8
-						node.param2 = node.param2 - orient +
-								wallmounted_substitution[orient + 1]
-						swap_node(pos, node)
+					if def.paramtype2 == "wallmounted" then
+						node.param2 = wallmounted_substitution[node.param2]
+						local meta = get_meta(pos):to_table()
+						set_node(pos, node)
+						get_meta(pos):from_table(meta)
```

Comment:
this is equivalent, no need to complicate it.
```suggestion
						swap_node(pos, node)
```